relay-server/src/services/outcome.rs

```diff
@@ -524,6 +527,146 @@ impl fmt::Display for DiscardReason {
     }
 }
 
+/// Similar to [`ItemType`] but it does not have any additional information in the
+/// Unknown variant so that it can derive [`Copy`] and be used from [`DiscardReason`].
+/// The variants should be the same as [`ItemType`].
+#[derive(Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, Clone)]
+pub enum DiscardItemType {
+    /// Event payload encoded in JSON.
+    Event,
+    /// Transaction event payload encoded in JSON.
+    Transaction,
+    /// Security report event payload encoded in JSON.
+    Security,
+    /// Raw payload of an arbitrary attachment.
+    Attachment,
+    /// Multipart form data collected into a stream of JSON tuples.
+    FormData,
+    /// Security report as sent by the browser in JSON.
+    RawSecurity,
+    /// NEL report as sent by the browser.
+    Nel,
+    /// Raw compressed UE4 crash report.
+    UnrealReport,
+    /// User feedback encoded as JSON.
+    UserReport,
+    /// Session update data.
+    Session,
+    /// Aggregated session data.
+    Sessions,
+    /// Individual metrics in text encoding.
+    Statsd,
+    /// Buckets of preaggregated metrics encoded as JSON.
+    MetricBuckets,
+    /// Client internal report (eg: outcomes).
+    ClientReport,
+    /// Profile event payload encoded as JSON.
+    Profile,
+    /// Replay metadata and breadcrumb payload.
+    ReplayEvent,
+    /// Replay Recording data.
+    ReplayRecording,
+    /// Replay Video data.
+    ReplayVideo,
+    /// Monitor check-in encoded as JSON.
+    CheckIn,
+    /// A log from the [OTEL Log format](https://opentelemetry.io/docs/specs/otel/logs/data-model/#log-and-event-record-definition)
+    OtelLog,
+    /// A log for the log product, not internal logs.
+    Log,
+    /// A standalone span.
+    Span,
+    /// A standalone OpenTelemetry span serialized as JSON.
+    OtelSpan,
+    /// An OTLP TracesData container.
+    OtelTracesData,
+    /// UserReport as an Event
+    UserReportV2,
+    /// ProfileChunk is a chunk of a profiling session.
+    ProfileChunk,
+    /// A new item type that is yet unknown by this version of Relay.
+    ///
+    /// By default, items of this type are forwarded without modification. Processing Relays and
+    /// Relays explicitly configured to do so will instead drop those items. This allows
+    /// forward-compatibility with new item types where we expect outdated Relays.
+    Unknown,
+    // Keep `Unknown` last in the list. Add new items above `Unknown`.
+}
+
+impl DiscardItemType {
+    pub fn as_str(&self) -> &'static str {
```

Comment:
Small doc comment would be nice on public methods (usually we have a linter for this).